lib/ansible/plugins/connection/vmware_tools.py

```diff
@@ -0,0 +1,369 @@
+from ssl import SSLEOFError, SSLError
+from time import sleep
+
+import requests
+import urllib3
+
+from ansible.errors import AnsibleError, AnsibleFileNotFound
+from ansible.module_utils._text import to_bytes, to_native
+from ansible.plugins.connection import ConnectionBase
+from ansible.utils.path import makedirs_safe
+
+try:
+    from pyVim.connect import Disconnect, SmartConnect, SmartConnectNoSSL
+    from pyVmomi import vim
+
+    HAS_PYVMOMI = True
+except ImportError as e:
+    HAS_PYVMOMI = False
+    PYVMOMI_IMPORT_ERROR = e
+
+
+__metaclass__ = type
+
+DOCUMENTATION = """
+    author: Deric Crago <deric.crago@gmail.com>
+    connection: vmware_tools
+    short_description: Execute modules via VMware Tools.
+    description:
+      - Execute modules via VMware Tools.
+      - "Note: Windows VMs will need to have C(ansible_shell_type: powershell) set."
+    version_added: "2.8"
+    requirements:
+      - pyvmomi (python library)
+    options:
+      connection_address:
+        description:
+          - Address for the connection (vCenter or ESXi Host).
+        vars:
+          - name: ansible_vmware_tools_connection_address
+        required: True
+      connection_username:
+        description:
+          - Username for the connection.
+          - "Requires the following permissions on the VM:
+               - VirtualMachine.GuestOperations.Execute
+               - VirtualMachine.GuestOperations.Modify
+               - VirtualMachine.GuestOperations.Query"
+        vars:
+          - name: ansible_vmware_tools_connection_username
+        required: True
+      connection_password:
+        description:
+          - Password for the connection.
+        vars:
+          - name: ansible_vmware_tools_connection_password
+        required: True
+      connection_verify_ssl:
+        description:
+          - Verify SSL for the connection.
+          - "Note: This will verify SSL for both C(connection_address) and the ESXi host running the VM."
+        vars:
+          - name: ansible_vmware_tools_connection_verify_ssl
+        default: True
+        type: bool
+      connection_ignore_ssl_warnings:
+        description:
+          - Ignore SSL warnings for the connection.
+        vars:
+          - name: ansible_vmware_tools_connection_ignore_ssl_warnings
+        default: False
+        type: bool
+      vm_path:
+        description:
+          - VM path relative to the connection.
+          - "vCenter Example: C(Datacenter/vm/Discovered virtual machine/testVM) (Needs to include C(vm) between the Datacenter and the rest of the VM path.)"
+          - "ESXi Host Example: C(ha-datacenter/vm/testVM) (Needs to include C(vm) between the Datacenter (C(ha-datacenter)) and the rest of the VM path.)"
+        vars:
+          - name: ansible_vmware_tools_vm_path
+        required: True
+      vm_username:
+        description:
+          - VM username.
+        vars:
+          - name: ansible_vmware_tools_vm_username
```

Comment:
be consistent with other connection plugins
```suggestion
          - name: ansible_vmware_tools_user
```

Reply:
`ansible_user` should win, I think, if it's defined, to be consistent w/ other plugins